### ISSUE TITLE
Strange while

### ISSUE THREAD

**Issue by MarcMazas** [javacc21/javacc21#58]
JavaCodeUtils.java, line 97 : 
        while (addToInvokedMethodList(invokedMethodNames, privateMethods));
Empty control-flow statement (the while loop has no block), so the logic is difficult to understand / maintain (repeated calls to method which keeps the same arguments).
At first glance the while loop should be in the called method.

**Comment by revusky** [comment]
Look, frankly, there is nothing wrong with this code. I don't even think it is particularly unclear. Well, okay, I grant the idiom is a bit unusual, but it is not a problem. We call `addToInvokeMethodList` repeatedly until it returns false. As far as I can tell, every direct alternative is more verbose:

      while (true) {
           if (addToInvokeMethodList(invokedMethodNames, privateMethods)) break;
      }

or defining even an intermediate variable to hold the result of the call:

        boolean result = true;
        while (result) result = addToInvokeMethodList(...);

The above is simply expressed more economically the way I wrote it:

        while (addToInvokeMethodList(...)) ;

Quite frankly, if Eclipse thinks there is something wrong with how I wrote the code, Eclipse can kiss my ass. Now, okay, it is useful to get warnings about *potential problems*, but to take this as seriously as you seem to do...

Really, such a tool has to be used with the appropriate *mind*. Like, you do have to understand that some of this stylistic advice that is algorithmically generated is just spurious! 

But the other thing I'm wondering about is what is the objective here? Are you bringing up these dubious stylistic suggestions in the code because you envisage doing some meaningful work on the code yourself? That would be great, but the answer to that question is in the negative, you're finally just trying to impose some empty formalisms on me, and though you surely do this with good intentions, it's just not very useful.
